### PR TITLE
Added tests for BatchNode

### PR DESCRIPTION
**Pull request by mattBrzezinski**
We currently have 0% code coverage on [src/batch_node.jl](https://codecov.io/gh/JuliaCloud/AWSClusterManagers.jl/tree/main/src) this is partially because the `Distributed.launch()` and `start_batch_node_worker()` functions are a bit tricky to test. However the other functions are quite easy to do, so I wrote some tests for them.

### CODE REVIEW

**Comment by mattBrzezinski** [comment]
bors try

File: test/batch_node.jl [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES -- String" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NUM_NODES" => "foobar"
```
Comment (mjram0s) [inline test/batch_node.jl:43]: Will this pass if it's a "1"?

File: test/batch_node.jl [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        @test AWSClusterManagers.parse_job_id(str) == case
+    end
+
+    failing_cases = [
+        string(repeat("a", 35), "#123"),
+        string(repeat("a", 37), "#123"),
+        string(repeat("!", 36), "#123"),
+        string(repeat("a", 36), "#"),
+        string(repeat("a", 36), "#abc"),
```
Comment (mjram0s) [inline test/batch_node.jl:100]: Would be helpful to add comments as to why these fail or why the other ones pass

File: test/batch_node.jl (hunk 1 above)
Comment (mattBrzezinski) [reply test/batch_node.jl:43]: Yes, we attempt to set the `num_workers` by `parse(Int, N)`. I can add in a test for this.

**Comment by mattBrzezinski** [comment]
bors try

**Review by mjram0s — APPROVED** [review]

**Pull request merged**